From d3b40463935ea1dbf6f26f6079a7d1e4988258a8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 7 Jan 2026 12:56:52 +0100
Subject: [PATCH 09/14] Build parts of hailort with C++ 17
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

abseil-cpp is a dependency of some parts of hailort and
headers in abseil-cpp 20250814 need C++ 17 as a minimum.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 hailort/hailort_service/CMakeLists.txt | 2 +-
 hailort/hailortcli/CMakeLists.txt      | 2 +-
 hailort/hrpc_protocol/CMakeLists.txt   | 2 +-
 hailort/libhailort/CMakeLists.txt      | 6 +++---
 hailort/libhailort/src/CMakeLists.txt  | 2 +-
 5 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/hailort/hailort_service/CMakeLists.txt b/hailort/hailort_service/CMakeLists.txt
index 1b1232d..72f5d89 100644
--- a/hailort/hailort_service/CMakeLists.txt
+++ b/hailort/hailort_service/CMakeLists.txt
@@ -15,7 +15,7 @@ add_executable(hailort_service
     ${HAILORT_SERVICE_OS_DIR}/hailort_service.cpp
 )
 target_compile_options(hailort_service PRIVATE ${HAILORT_COMPILE_OPTIONS})
-set_property(TARGET hailort_service PROPERTY CXX_STANDARD 14)
+set_property(TARGET hailort_service PROPERTY CXX_STANDARD 17)
 target_link_libraries(hailort_service
     libhailort
     hailort_common
diff --git a/hailort/hailortcli/CMakeLists.txt b/hailort/hailortcli/CMakeLists.txt
index 0c48f0b..961d271 100644
--- a/hailort/hailortcli/CMakeLists.txt
+++ b/hailort/hailortcli/CMakeLists.txt
@@ -61,7 +61,7 @@ add_executable(hailortcli
 )
 
 target_compile_options(hailortcli PRIVATE ${HAILORT_COMPILE_OPTIONS})
-set_property(TARGET hailortcli PROPERTY CXX_STANDARD 14)
+set_property(TARGET hailortcli PROPERTY CXX_STANDARD 17)
 
 # Link with a relative libhailort
 set_property(TARGET hailortcli PROPERTY INSTALL_RPATH "$ORIGIN")
diff --git a/hailort/hrpc_protocol/CMakeLists.txt b/hailort/hrpc_protocol/CMakeLists.txt
index 1553992..6d492c9 100644
--- a/hailort/hrpc_protocol/CMakeLists.txt
+++ b/hailort/hrpc_protocol/CMakeLists.txt
@@ -5,7 +5,7 @@ get_filename_component(PROTO_HEADER_DIRECTORY ${PROTO_RPC_HEADER} DIRECTORY)
 
 add_library(rpc_proto STATIC EXCLUDE_FROM_ALL ${PROTO_RPC_SRC} ${PROTO_RPC_HEADER})
 target_link_libraries(rpc_proto protobuf-lite)
-set_target_properties(rpc_proto PROPERTIES CXX_STANDARD 14 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
+set_target_properties(rpc_proto PROPERTIES CXX_STANDARD 17 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
 if(CMAKE_HOST_WIN32)
     # https://github.com/protocolbuffers/protobuf/tree/master/cmake#notes-on-compiler-warnings
     target_compile_options(rpc_proto PRIVATE /wd4244)
diff --git a/hailort/libhailort/CMakeLists.txt b/hailort/libhailort/CMakeLists.txt
index 3680122..544dc7b 100644
--- a/hailort/libhailort/CMakeLists.txt
+++ b/hailort/libhailort/CMakeLists.txt
@@ -15,7 +15,7 @@ protobuf_generate_python(PROTO_HEF_PY tracer_profiler.proto)
 
 add_library(hef_proto ${PROTO_HEF_SRC} ${PROTO_HEF_HEADER} ${PROTO_HEF_PY})
 target_link_libraries(hef_proto protobuf-lite)
-set_target_properties(hef_proto PROPERTIES CXX_STANDARD 14 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
+set_target_properties(hef_proto PROPERTIES CXX_STANDARD 17 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
 if(CMAKE_HOST_WIN32)
     # https://github.com/protocolbuffers/protobuf/tree/master/cmake#notes-on-compiler-warnings
     target_compile_options(hef_proto PRIVATE /wd4244)
@@ -30,7 +30,7 @@ target_include_directories(hef_proto
 protobuf_generate_cpp(PROTO_SCHEDULER_MON_SRC PROTO_SCHEDULER_MON_HEADR scheduler_mon.proto)
 add_library(scheduler_mon_proto ${PROTO_SCHEDULER_MON_SRC} ${PROTO_SCHEDULER_MON_HEADR})
 target_link_libraries(scheduler_mon_proto protobuf-lite)
-set_target_properties(scheduler_mon_proto PROPERTIES CXX_STANDARD 14 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
+set_target_properties(scheduler_mon_proto PROPERTIES CXX_STANDARD 17 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
 if(CMAKE_HOST_WIN32)
     target_compile_options(scheduler_mon_proto PRIVATE /wd4244)
 endif()
@@ -44,7 +44,7 @@ target_include_directories(scheduler_mon_proto
 protobuf_generate_cpp(PROTO_PROFILER_SRC PROTO_PROFILER_HEADR tracer_profiler.proto)
 add_library(profiler_proto ${PROTO_PROFILER_SRC} ${PROTO_PROFILER_HEADR})
 target_link_libraries(profiler_proto protobuf-lite)
-set_target_properties(profiler_proto PROPERTIES CXX_STANDARD 14 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
+set_target_properties(profiler_proto PROPERTIES CXX_STANDARD 17 GENERATED TRUE POSITION_INDEPENDENT_CODE ON)
 if(CMAKE_HOST_WIN32)
     target_compile_options(profiler_proto PRIVATE /wd4244)
 endif()
diff --git a/hailort/libhailort/src/CMakeLists.txt b/hailort/libhailort/src/CMakeLists.txt
index 26461f8..7ea72aa 100644
--- a/hailort/libhailort/src/CMakeLists.txt
+++ b/hailort/libhailort/src/CMakeLists.txt
@@ -138,7 +138,7 @@ set_target_properties(libhailort PROPERTIES
     VERSION ${HAILORT_MAJOR_VERSION}.${HAILORT_MINOR_VERSION}.${HAILORT_REVISION_VERSION}
     # SOVERSION ${HAILORT_MAJOR_VERSION}
 
-    CXX_STANDARD              14
+    CXX_STANDARD              17
     CXX_STANDARD_REQUIRED     YES
     CXX_EXTENSIONS            NO
     C_VISIBILITY_PRESET       hidden
-- 
2.52.0

